### ISSUE TITLE
Device ID and IMEI spoofing

### ISSUE THREAD

**Issue by b-irl** [copperhead/bugtracker#330]
It'd be nice to selectively or collectively deny apps the ability to uniquely identify and report information about devices.

Hide or spoof the device ID/IMEI, especially from apps
 -- Ideally, each app would see a unique device ID/IMEI, which may be randomized once or each time the app started. There are apps for rooted devices, like XPrivacy, which offer features such as this.

Of course, when installing a new app, the default should be no permissions at all, too.

Thanks for MAC spoofing by default!!!

***\* NOTE: The below was removed following discussion below.
-- Ideally, the wireless network would see a unique IMEI for each SIM card inserted. For example, the first time you insert SIM A, a unique IMEI is generated for it. The first time you insert SIM B, a unique IMEI is generated for it. When you switch between the SIM cards, the network sees a consistant IMEI per SIM card.

**Comment by thestinger** [comment]
> -- Ideally, the wireless network would see a unique IMEI for each SIM card inserted. For example, the first time you insert SIM A, a unique IMEI is generated for it. The first time you insert SIM B, a unique IMEI is generated for it. When you switch between the SIM cards, the network sees a consistant IMEI per SIM card.

That's not in-scope for CopperheadOS as it exists today.

**Comment by b-irl** [comment]
I was under the impression that people use prepaid phones, rather than swapping pre-paid SIM cards, due to the fact that change SIM cards alone will allow the carrier to corrolate SIM cards with eachother via a hard-coded IMEI.

**Comment by thestinger** [comment]
Sure, but we aren't making our own phone hardware. It's a software-only project with one developer for the time being. If it's not something in Android, then it's out-of-scope. It's not possible to change the lower level firmware.

**Comment by b-irl** [comment]
I've adjusted the original enhancement request to remove the impossible

**Comment by thestinger** [comment]
Thanks, I would like to get to the point where we could stuff like that too, but for the foreseeable future it's impossible.

**Comment by asdf-asdf** [comment]
Here's an article on how you can do it manually...no idea if this actually works or not as I haven't tried it yet - 2spooky4me
http://www.hugestreet.info/2015/08/Free-Android-Device-ID-and-IMEI-number-changer.html

The biggest reason for me not doing this myself yet is that I'm not sure of the legality of doing this...I'm guessing this is comparable to changing your MAC address - which is completely legal - but I honestly have no idea about this. I'm pretty sure having the same IMEI on a single network is not a good idea. If anyone has any insight as to the legality of changing IMEI's in the US or Canada, that'd be awesome to know.

I also don't know of how detectable this would be to the phone companies. As the article suggests; it's probably best not to use the same dev ID, so I would take out your SIM card and factory reset your phone before if you're going to do this. Maybe reset it again afterwards for safe measure :)

If you're going to change your IMEI, you may want to go to https://imeidata.net/blacklist/check or something comparable to ensure you don't set it with a broken IMEI. Most phone companies also have a tool to check IMEI's on their websites.

tl;dr
```
1. First Dial *#7465625# or *#*#3646633#*#* on your android device.
2. Now, click on Connectivity option or call pad,
3. Now look for CDS information and then click on it.
4. Then, checkout for Radio information.
5. Now, if your android device is a dual sim device. Then you will get two option like IMEI_1  [SIM1] and IMEI_2 [SIM2]. Then select any one or of which you want to change IMEI number.
6. Now, change your IMEI number illustrated below,
             AT +EGMR=1,7,”IMEI_1” and “AT +EGMR=1,10,”IMEI_2”
7. Now, just replace IMEI1 or IMEI2 with your any your desired number. and then click on SEND.
```

**Comment by thestinger** [comment]
ANDROID_ID is changing in Android O so I'm delaying any planning of changes until after that has happened: https://developer.android.com/preview/behavior-changes.html#privacy-all.
